wscript

```diff
@@ -113,11 +120,21 @@ def download_and_extract(destdir, url, ext):
       print('Found tarball at {0}'.format(dest))
 
     print('Extracting {0}'.format(dest))
-    # TODO: make portable.
+
+    # Unpacking using 7z do not guarantee to work for various reasons, see #505: https://github.com/cquery-project/cquery/issues/505
+    # To overcome this situation it is better to let the installer do its jobs but silently.
+    # For more info on silent mode: http://nsis.sourceforge.net/Docs/Chapter4.html#silent
     if ext == '.exe':
-      subprocess.call(['7z', 'x', '-o{0}'.format(destdir), '-xr!$PLUGINSDIR', dest])
+      try:
+        output_dir = os.path.abspath(os.path.join(out, CLANG_TARBALL_NAME))
+        subprocess.check_call([dest, '/S', '/D=%s' % output_dir])
+      except subprocess.CalledProcessError as e:
+        print('[rc=%d] Failed to unpack clang compiler' % e.returncode)
     else:
-      subprocess.call(['tar', '-x', '-C', out, '-f', dest])
+      with xz.open(dest, 'rb') as xz_file:
```

Comment:
I'd prefer falling back to tar on python2 instead of requiring a pip install.

Reply:
I can understand that. What's tought about the .exe section ?

Reply:
I am not sure `.exe` can be unpacked with the new code.

Reply:
Isn't the Python way much slower than the external  `tar` process?

Reply:
Well I tried on my personal computer under Windows 7 and unpacking the .exe work just fine, but I am a mitigated with this solution. On one side we just use the installer as it first purpose. On the other side UAC and other weird stuff can happen.

My suggestion is: like the current wscript try to unpack using 7zip. If it fail, let the user choose between these 2 options: silent installation or let him unpack by himself. What you think about this ?

Reply:
How can "silent installation" work? Let the user unpack by himself seems good